### PR TITLE
Add support for building Kali image

### LINKED ISSUE #31: Use InjectMiniccc API function when include_miniccc is enabled in image config
---

Originally created by @activeshadow on *Fri, 26 Feb 2021 20:43:39 GMT*

---
Also consider adding API functions and CLI commands for injecting protonuke and minirouter into images.

### PR DESCRIPTION
**Pull request by activeshadow**
* Added default image config for Kali
* Updated Dockerfile to include Kali Archive keyring

This commit also includes some updates to image building in general.

* Print deprecation warning for inject_miniccc and inject_protonuke
* Added inject-miniexe subcommand for miniccc, protonuke, and minirouter
* Deprecated inject-miniccc in favor of inject-miniexe

closes #31

### CODE REVIEW

File: src/go/cmd/image.go [hunk 1]
```diff
@@ -456,40 +455,57 @@ func newImageUpdateCmd() *cobra.Command {
...
+	}
 
-	In a Linux disk image, the miniccc agent will be injected into
+	return cmd
+}
+
+func newImageInjectMiniExeCmd() *cobra.Command {
+	desc := `Inject a minimega executable into a disk image
+
+	Used to add a minimege executable (miniccc, protonuke, or minirouter) and
```
Comment (glattercj) [inline src/go/cmd/image.go:475]: `minimege`...sounds fancy

File: src/go/api/image/image.go [hunk 2]
```diff
@@ -520,58 +521,72 @@ func InjectMiniccc(agent, disk, svc string) error {
...
+			exe + fmt.Sprintf(":/minimega/%s.exe", base),
 		}
-	} else { // assume Linux
-		if err := os.MkdirAll(tmp+"/miniccc/symlinks", 0755); err != nil {
+	} else {
+		// /opt/minimega/bin/miniccc --> miniccc
+		base := path.Base(exe)
+
+		if base != "miniccc" && base != "protonuke" && base != "minirouter" {
+			return fmt.Errorf("only miniccc, protonuke, and minirouter is supported for Linux")
```
Comment (glattercj) [inline src/go/api/image/image.go:545]: s/is/are

File: src/go/api/image/image.go [hunk 3]
```diff
@@ -520,58 +521,72 @@ func InjectMiniccc(agent, disk, svc string) error {
...
 
-	if path.Ext(agent) == ".exe" { // assume Windows
-		if err := tmpl.RestoreAsset(tmp, "miniccc/miniccc-scheduler.cmd"); err != nil {
-			return fmt.Errorf("restoring miniccc scheduler for Windows: %w", err)
+	if path.Ext(exe) == ".exe" { // assume Windows
+		// /opt/minimega/bin/miniccc.exe --> miniccc
+		base := strings.TrimSuffix(path.Base(exe), path.Ext(exe))
+
+		if base != "miniccc" && base != "protonuke" {
+			return fmt.Errorf("only miniccc.exe and protonuke.exe is supported for Windows")
```
Comment (glattercj) [inline src/go/api/image/image.go:529]: s/is/are

**Comment by activeshadow** [comment]
@glattercj take another look when you have a sec. Thoughts on how I handled protonuke for Windows would be appreciated.

**Review by glattercj — APPROVED** [review]

**Pull request merged**